src/pybind/mgr/cephadm/serve.py

```diff
@@ -59,6 +59,8 @@ def serve(self) -> None:
         of cephadm. This loop will then attempt to apply this new state.
         """
         self.log.debug("serve starting")
+        self.mgr.config_checker.load_network_config()
```

Comment:
Guess you have to run load_network_config before every call to `config_checker.run_checks`. Those network configs might change at runtime.